fvm/evm/stdlib/contract.go

```diff
@@ -97,6 +99,48 @@ func (e abiDecodingError) Error() string {
 	return b.String()
 }
 
+func calculateComputation(
+	inter *interpreter.Interpreter,
+	values *interpreter.ArrayValue,
+	evmAddressTypeID common.TypeID,
+) int {
+	computation := 0
+
+	values.Iterate(inter, func(element interpreter.Value) (resume bool) {
+		switch value := element.(type) {
+		case *interpreter.StringValue:
+			// Dynamic variables, such as strings, are encoded
+			// in 3 chunks of 32 bytes. The first chunk contains
+			// the index where information for the string begin,
+			// the second chunk contains the number of bytes the
+			// string occupies, and the third chunk contains the
+			// value of the string itself.
+			computation += 3 * abiEncodingByteSize
+		case *interpreter.CompositeValue:
+			if value.TypeID() == evmAddressTypeID {
```

Comment:
what if this is not true?

Reply:
Maybe also explicitly report an error for unsupported types

Reply:
Updated in https://github.com/onflow/flow-go/pull/5146/commits/32e6b6f0b533f8de085671e5fff80c4e4587118f